resources/assets/js/src/components/chart/Chart.tsx

```diff
@@ -101,7 +101,11 @@ export default class Chart extends React.Component<IProps, IState> {
                             </LineChart>
                         </ResponsiveContainer>
                     </Swipeable>
-                    <ButtonPanel clickHandler={this.refreshDataByEventHandler} />
+                    <ButtonPanel
+                        clickHandler={this.refreshDataByEventHandler}
+                        initialDate={this.state.initialDate}
+                        lastDate={this.state.data[this.state.data.length - 1].date}
```

Comment:
do not handle undefined

Reply:
what do you mean?

Reply:
if you get empty data array, array index will be out of bounds and you will get undefined

Reply:
if I get empty array, ButtonPanel won't be initialized